src/Features/Core/Portable/ReplaceConditionalWithStatements/AbstractReplaceConditionalWithStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,341 @@
+    protected abstract bool CanRewriteLocalDeclarationStatement(TLocalDeclarationStatementSyntax localDeclarationStatement);
+
+    protected abstract TLocalDeclarationStatementSyntax GetUpdatedLocalDeclarationStatement(SyntaxGenerator generator, TLocalDeclarationStatementSyntax localDeclarationStatement, ILocalSymbol symbol);
+
+    private bool IsSupportedSimpleStatement([NotNullWhen(true)] TStatementSyntax? statement)
+        => IsAssignmentStatement(statement) ||
+           statement is TExpressionStatementSyntax or
+                        TReturnStatementSyntax or
+                        TThrowStatementSyntax or
+                        TYieldStatementSyntax;
+
+    public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+    {
+        var (document, _, cancellationToken) = context;
+
+        // First, see if we're inside a conditional.  If so, attempt to use that to offer fixes on.
+        var conditionalExpression = await context.TryGetRelevantNodeAsync<TConditionalExpressionSyntax>().ConfigureAwait(false);
+        if (conditionalExpression is not null)
+        {
+            TryHandleConditionalExpression(context, conditionalExpression);
+            return;
+        }
+
+        // If not, see if we're on an supported statement.  If so, see if it has an applicable conditional within it
+        // that we could support this on.
+        var statement = await context.TryGetRelevantNodeAsync<TStatementSyntax>().ConfigureAwait(false);
+        if (IsSupportedSimpleStatement(statement))
+        {
+            TryHandleConditionalExpression(context, TryFindConditional(statement, cancellationToken));
+            return;
+        }
+
+        if (statement is TLocalDeclarationStatementSyntax localDeclarationStatement)
+        {
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var variables = syntaxFacts.GetVariablesOfLocalDeclarationStatement(localDeclarationStatement);
+            if (variables.Count == 1)
+            {
+                TryHandleConditionalExpression(
+                    context, TryFindConditional(syntaxFacts.GetInitializerOfVariableDeclarator(variables[0]), cancellationToken));
+                return;
+            }
+        }
+    }
+
+    private static TConditionalExpressionSyntax? TryFindConditional(SyntaxNode? top, CancellationToken cancellationToken)
+    {
+        return Recurse(top);
+
+        TConditionalExpressionSyntax? Recurse(SyntaxNode? node)
+        {
+            if (node is not null)
+            {
+                foreach (var child in node.ChildNodesAndTokens())
+                {
+                    if (child.IsToken)
+                        continue;
+
+                    var result = CheckNode(child.AsNode());
+                    if (result != null)
+                        return result;
+                }
```

Comment:
```suggestion
                foreach (var child in node.ChildNodes())
                {
                    var result = CheckNode(child);
                    if (result != null)
                        return result;
                }
```

Reply:
ChildNodes allocates, while ChildNodesAndTokens does not :)

Reply:
#wontfix

Reply:
Thanks for clarifying. Nice to know.